erpnext/regional/india/e_invoice/utils.py

```diff
@@ -43,8 +43,9 @@ def validate_eligibility(doc):
 	invalid_supply_type = doc.get('gst_category') not in ['Registered Regular', 'SEZ', 'Overseas', 'Deemed Export']
 	company_transaction = doc.get('billing_address_gstin') == doc.get('company_gstin')
 	no_taxes_applied = not doc.get('taxes')
+	has_non_gst_item = any(d for d in doc.get('items') if frappe.db.get_value('Item', d.get('item_code'), 'is_non_gst'))
```

Comment:
`is_non_gst` check is also maintained at the Item child tables which is fetched from item master, you can directly check in the child table instead of querying Item Master

Also what if any invoice has both GST and Non GST items, not sure if this is a valid scenario

Reply:
Yeah, will fetch it from child table.